packages/fetch-engine/src/download.ts

```diff
@@ -291,19 +291,8 @@ async function binaryNeedsToBeDownloaded(
     }
   }
 
-  // If there is no cache and the file doesn't exist, we for sure need to download it
-  if (!targetExists) {
-    debug(`file ${job.targetFilePath} does not exist and must be downloaded`)
-    return true
-  }
-
-  // 3. If same platform, always check --version
-  if (job.binaryTarget === nativePlatform && job.binaryName !== BinaryType.libqueryEngine) {
```

Comment:
- Then we land here, and we mostly use `library` engine
- That also means that we skip this and continue below